src/lib/utils/index.ts

```diff
@@ -337,3 +337,45 @@ export function createContextualError<Context extends Record<string, unknown>>(
 
   return e
 }
+
+export type SerializedError = {
+  name: string
+  message: string
+  stack?: string
+} & JsonObject
+
+export function serializeError(e: Error): SerializedError {
+  return {
+    name: e.name,
+    message: e.message,
+    stack: e.stack,
+    ...e,
+  }
+}
+
+export function deserializeError(se: SerializedError): Error {
```

Comment:
Unfortunately serializeError lib we're using is not reliable, I had to log the error myself to see what was going on. See how the serialize error leads to `{}`. Maybe b/c its a TypeError instead of vanilla Error

![image (1)](https://user-images.githubusercontent.com/284476/82107475-01878000-96f6-11ea-8aba-be2e2dabe2d6.png)